src/rust/src/lib.rs

```diff
@@ -2,7 +2,84 @@
 // 2.0, and the BSD License. See the LICENSE file in the root of this repository
 // for complete details.
 
+/// Returns the value of the input with the most-significant-bit copied to all
+/// of the bits.
+fn duplicate_msb_to_all(a: u8) -> u8 {
+    0u8.wrapping_sub(a >> 7)
+}
+
+/// This returns 0xFF if a < b else 0x00, but does so in a constant time
+/// fashion.
+fn constant_time_lt(a: u8, b: u8) -> u8 {
+    duplicate_msb_to_all(a ^ ((a ^ b) | ((a.wrapping_sub(b)) ^ b)))
+}
```

Comment:
Nit: unnecessary braces
```suggestion
fn constant_time_lt(a: u8, b: u8) -> u8 {
    duplicate_msb_to_all(a ^ ((a ^ b) | (a.wrapping_sub(b) ^ b)))
}
```